astropy/io/misc/hdf5.py

```diff
@@ -213,6 +211,21 @@ def _encode_mixins(tbl):
 
     return encode_tbl
 
+def check_column_encoding(table):
+    """
+    Table with native py3 strings can't be stored in a hdf5 format so if
+    table has py3 strings then a copy of table is made and stored in temp.
+    the copy stored in temp has columns in bytes unicode.
+    """
+    col=[column for column in table.colnames if table.columns[column].info.dtype.kind=='U']
```

Comment:
I think usually in the astropy code, `col` or `column` refer to a `Column` object, so the variable names do not describe well what they contain. You are creating of list of *column names* and iterating on *names*, so I would use something like `colnames = [name for name in ...]` and the same below.